public/app/ui/components/search/basic/BasicSearchComponent.js

```diff
@@ -0,0 +1,111 @@
+(function () {
+    'use strict';
+
+    define(
+        [
+            'lodash',
+            'knockout'
+        ],
+        function (_, ko) {
+            return function (parameters) {
+                var selectedSearchFieldKey, initialFieldAndText,
+                    getFieldAndTextForQuery = function (query) {
+                        var selectedKey, selectedField;
+                        if (!query || !query.children || query.children.length === 0) {
+                            return undefined;
+                        }
+                        selectedKey = query.children[0].field;
+                        if (!selectedKey) {
+                            return undefined;
+                        }
+                        selectedField = _.find(parameters.fields(), { key: selectedKey });
+                        if (!selectedField || !selectedField.basicSearch) {
+                            return undefined;
+                        }
+                        return {
```

Comment:
Hmmm, charitable of you, but using "and" in a variable/function name is a bad code smell.  But it works and is only internal to this one module, so it's good enough :-)